sdk/eventhub/Azure.Messaging.EventHubs/MIGRATIONGUIDE.md

```diff
@@ -0,0 +1,380 @@
+
+The code below assumes all events fit into a single batch. For more complete example, see sample: [Publish events
+to specific partition](https://github.com/Azure/azure-sdk-for-net/blob/master/sdk/eventhub/Azure.Messaging.EventHubs/samples/Sample06_PublishAnEventBatchToASpecificPartition.cs).
+
+So in v4:
+```csharp
+var connectionString = "<< CONNECTION STRING FOR THE EVENT HUBS NAMESPACE >>";
+var eventHubName = "<< NAME OF THE EVENT HUB >>";
+
+var eventHubClient = EventHubClient.CreateFromConnectionString(connectionString, eventHubName))
+PartitionSender partitionSender  =  eventHubClient.CreatePartitionSender("my-partition-id");
+try
+{
+    EventDataBatch eventBatch = partitionSender.CreateBatch();
+    eventBatch.TryAdd(new EventData(Encoding.UTF8.GetBytes("First")));
+    eventBatch.TryAdd(new EventData(Encoding.UTF8.GetBytes("Second")));
+
+    await partitionSender.SendAsync(eventBatch);
+}
+catch()
+{
+    // Handle the exception here
+}
+finally
+{
+    await eventHubClient.CloseAsync();
+    await partitionSender.CloseAsync();
+}
+```
+
+In v5:
+```csharp
+var connectionString = "<< CONNECTION STRING FOR THE EVENT HUBS NAMESPACE >>";
+var eventHubName = "<< NAME OF THE EVENT HUB >>";
+
+await using (var producerClient = new EventHubProducerClient(connectionString, eventHubName))
+{
+    var batchOptions = new CreateBatchOptions() { PartitionId = "my-partition-id" };
+    using EventDataBatch eventBatch = await producerClient.CreateBatchAsync(batchOptions);
+    eventBatch.TryAdd(new EventData(Encoding.UTF8.GetBytes("First")));
+    eventBatch.TryAdd(new EventData(Encoding.UTF8.GetBytes("Second")));
+    
+    await producerClient.SendAsync(eventBatch);
+}
+```
+
+### Migrating code from `EventHubClient` to `EventHubProducerClient` for publishing events using automatic routing
+
+In v4, events could be published to an Event Hub that allowed the service to automatically route events to an available partition.
+
+In v5, automatic routing occurs when an [`EventDataBatch`](https://docs.microsoft.com/en-us/dotnet/api/azure.messaging.eventhubs.eventdatabatch?view=azure-dotnet-preview) is created using [`CreateBatchAsync()`](https://docs.microsoft.com/en-us/dotnet/api/azure.messaging.eventhubs.eventhubproducerclient.createbatchasync?view=azure-dotnet-preview#Azure_Messaging_EventHubs_EventHubProducerClient_CreateBatchAsync_System_Threading_CancellationToken_).
+
+So in v4:
+```csharp
+var connectionString = "<< CONNECTION STRING FOR THE EVENT HUBS NAMESPACE >>";
+var eventHubName = "<< NAME OF THE EVENT HUB >>";
+
```

Comment:
Do you think we should include samples around the single event and set of event cases too?

Reply:
I believe, `In v4 you could also send a single event, a set of events, or an explicit batch. We are only supporting batch currently to ensure that there are no unexpected exceptions generated during send; you can't put it in a batch if it was too large to send.` information is sufficient for the guide([Java ](https://github.com/Azure/azure-sdk-for-java/blob/master/sdk/eventhubs/azure-messaging-eventhubs/migration-guide.md)and [JS ](https://github.com/Azure/azure-sdk-for-js/blob/master/sdk/eventhub/event-hubs/migrationguide.md)also didn't add these samples in their guide).

If you feel like we should include these samples, we can add these later :)

Reply:
We have mentioned the following in our migration doc:

`In v4 you could also send a single event, a set of events, or an explicit batch. We are only supporting batch currently to ensure that there are no unexpected exceptions generated during send; you can't put it in a batch if it was too large to send.` 

And, I believe this information is sufficient for the guide([Java ](https://github.com/Azure/azure-sdk-for-java/blob/master/sdk/eventhubs/azure-messaging-eventhubs/migration-guide.md)and [JS ](https://github.com/Azure/azure-sdk-for-js/blob/master/sdk/eventhub/event-hubs/migrationguide.md)also didn't add these samples in their guide).

If you feel like we should include these samples, we can add these later :)

Reply:
I don't feel strongly either way.    Most of the commentary here is stream-of-consciousness and potentially not fully considered  😄